### PR TITLE
[Issue 5922] - Cleanup poms to use ${project.groupId} and ${project.version} consistently

### LINKED ISSUE #5922: On the dependencies in the pom.xml files.
**Describe the bug**
Most of pom.xml files, in which some dependencies on the pulsar itself are written as following:
    ......
    \<dependency>
      \<groupId>org.apache.pulsar\</groupId> // **THIS LINE**
      \<artifactId>pulsar------\</artifactId>
      \<version>${project.version}\</version>
    \</dependency>
    ......
But most of them are like this:
\<dependency>
      \<groupId>${project.groupId}\</groupId> // **THIS LINE**
      \<artifactId>pulsar------\</artifactId>
      \<version>${project.version}\</version>
    \</dependency>

**Although both of them work, why not use JUST one of them (e.g.  \<groupId>${project.groupId}\</groupId>)**. 
pulsar-pom

### PR DESCRIPTION
**Pull request by vcottagiri**
Fixes #5922

### Motivation
As described in #5922 
To make the poms consistent across the codebase

### Modifications
- Replaced <groupId>org.apache.pulsar</groupId> to <groupId>${project.groupId}</groupId> where applicable
- Replaced <version>2.7.0-SNAPSHOT</version> to <version>${project.version}</version> where applicable
- Replaced <version>${project.parent.version}</version> to <version>${project.version}</version> where applicable
- Removed <groupId>org.apache.pulsar</groupId> when redundant

### Verifying this change

- [X] Make sure that the change passes the CI checks.
- [X] This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

File: examples/flink/pom.xml [hunk 1]
```diff
@@ -74,6 +72,12 @@
       <version>${project.version}</version>
     </dependency>
 
+    <dependency>
```
Comment (vcottagiri) [inline examples/flink/pom.xml:75]: @merlimat 
Sorry, this dependency is not needed as its inherited from the root pom.
I will create a PR fixing this asap

Comment (merlimat) [reply examples/flink/pom.xml:75]: Ok, the buildtools is anyway injected as a dependency for all the modules

Comment (vcottagiri) [reply examples/flink/pom.xml:75]: Yes, my bad

Comment (vcottagiri) [reply examples/flink/pom.xml:75]: fixed by #7560

**Pull request merged**